src/notelistmodel.cpp

```diff
@@ -384,13 +384,11 @@ bool NoteListModel::dropMimeData(const QMimeData *mime, Qt::DropAction action, i
                                  const QModelIndex &parent)
 {
     Q_UNUSED(column);
-    auto getRef = [this](int row) -> NodeData & {
-        if (row < m_pinnedList.size()) {
-            return m_pinnedList[row];
-        } else {
-            row = row - m_pinnedList.size();
-            return m_noteList[row];
+    auto getRef = [this](int rowIndex) -> NodeData & {
+        if (rowIndex < m_pinnedList.size()) {
+            return m_pinnedList[rowIndex];
         }
+        return m_noteList[rowIndex - m_pinnedList.size()];
```

Comment:
You're changing one of 6 occurrences of this `getRef` function. Consider changing it to a private member and replacing all occurrences.

Reply:
I tried to implement this, but I don't think my way is the "correct" way of doing it, so it'd be great if you could take a look. Thanks!